kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/ParquetSchemaUtils.java

```diff
@@ -186,6 +186,43 @@ private static Type prunedType(Type type, DataType deltaType) {
       GroupType groupType = (GroupType) type;
       StructType structType = (StructType) deltaType;
       return groupType.withNewFields(pruneFields(groupType, structType));
+    } else if (type instanceof GroupType && deltaType instanceof ArrayType) {
```

Comment:
Can you add a comment [here](https://github.com/delta-io/delta/blob/3800ac40a0d3cccd97000862880ba5df67dd8e2d/kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/ArrayColumnReader.java#L65) to update this pruning code if we add support for other parquet array schema structures?